### PR TITLE
Claimables [PR #3]: claim functionality

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-1786

## What changed (plus any additional context for devs)
added functionality to claim sponsored and transaction type claimables

## Screen recordings / screenshots

https://github.com/user-attachments/assets/4971c143-9b71-4205-b3c1-9509a9a1ce19

https://www.loom.com/share/1ff82050845045debfd0dd872b9ddd2f

## What to test
first turn on the feature flag in settings

if you have your own wallet w/ claimables, test out claiming those

1. add notben.eth as a wallet
2. set `enableActionsOnReadOnlyWallet` to true in src/config/debug.ts
3. claim sponsored/transaction type claimables from wallet screen claimables section

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@benisgold/claim-functionality/e2850d89ba07346f1b1a5a8b1373d2534923521c.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@benisgold/claim-functionality/e2850d89ba07346f1b1a5a8b1373d2534923521c.ipa) for e2850d89ba07346f1b1a5a8b1373d2534923521c

File: src/screens/claimables/ClaimingClaimableSharedUI.tsx [hunk 1]
```diff
@@ -216,9 +218,13 @@ export const ClaimingClaimableSharedUI = ({
                   }
                 }}
                 onLongPress={() => {
-                  if (claimStatus === 'idle' || claimStatus === 'error') {
-                    setClaimStatus('claiming');
-                    claim();
+                  if (!isReadOnlyWallet || enableActionsOnReadOnlyWallet) {
+                    if (claimStatus === 'idle' || claimStatus === 'error') {
+                      setClaimStatus('claiming');
```
Comment (greg-schrammel) [inline src/screens/claimables/ClaimingClaimableSharedUI.tsx:223]: I think we don't need this `setClaimStatus`, the `useMutation` has a `status`, we could hoist this logic of alerting on readOnly wallets to the parent and use this as just a presentation component

File: src/screens/claimables/ClaimingSponsoredClaimable.tsx [hunk 2]
```diff
@@ -1,9 +1,88 @@
...
+      const provider = getProvider({ chainId: claimable.chainId });
+      const wallet = await loadWallet({
+        address: accountAddress,
+        showErrorIfNotLoaded: true,
+        provider,
+      });
+
+      if (!wallet) {
+        // Biometrics auth failure (retry possible)
+        setClaimStatus('error');
```
Comment (greg-schrammel) [inline src/screens/claimables/ClaimingSponsoredClaimable.tsx:30]: wdyt of throwing here and using `status` from `useMutation`?

File: src/screens/claimables/ClaimingTransactionClaimable.tsx [hunk 3]
```diff
@@ -157,7 +158,7 @@ export const ClaimingTransactionClaimable = ({ claimable }: { claimable: Transac
 
       const { errorMessage } = await walletExecuteRap(wallet, {
```
Comment (estebanmino) [inline src/screens/claimables/ClaimingTransactionClaimable.tsx:159]: am i lost or this is using current raps logic

i was just reviewing thi https://github.com/rainbow-me/rainbow/pull/6138

Comment (benisgold) [reply src/screens/claimables/ClaimingTransactionClaimable.tsx:162]: it uses v2

File: src/screens/claimables/ClaimingTransactionClaimable.tsx [hunk 4]
```diff
@@ -134,9 +139,61 @@ export const ClaimingTransactionClaimable = ({ claimable }: { claimable: Transac
...
+
+      if (!wallet) {
+        // Biometrics auth failure (retry possible)
+        setClaimStatus('error');
+        return;
+      }
+
+      const { errorMessage } = await walletExecuteRap(wallet, {
+        type: 'claimTransactionClaimableRap',
+        claimTransactionClaimableActionParameters: { claimTx: txPayload, asset: claimable.asset },
```
Comment (benisgold) [inline src/screens/claimables/ClaimingTransactionClaimable.tsx:164]: @walmat you already reviewed this fn in previous PR, it was actually meant to be added in this PR so i moved it here. the only difference is this new param `asset` that's passed to the walletExecuteRap. it's used to generate the pending tx.

File: src/rapsV2/actions/claimTransactionClaimableAction.ts [hunk 5]
```diff
@@ -7,39 +7,37 @@ import { NewTransaction } from '@/entities';
...
+  }
 
-  // if (!result?.result || !!result.error || !result.result.hash) {
-  //   throw new RainbowError('[CLAIM-TRANSACTION-CLAIMABLE]: failed to execute claim transaction');
-  // }
+  const transaction = {
+    amount: '0x0',
+    gasLimit: result.result.gasLimit,
+    from: result.result.from ?? null,
+    to: result.result.to ?? null,
```
Comment (walmat) [inline src/rapsV2/actions/claimTransactionClaimableAction.ts:23]: should we fail if to/from are null? That basically is sending code to a new contract I think

**Review by walmat — APPROVED** [review]

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**